### PR TITLE
WIP: New Achivement and Badge: Joined a Guild

### LINKED ISSUE #8614: New Achievement and Badge: Joined a Guild
A new achievement and badge are ready to be implemented! This should be received when a player joins their first guild. The badge image and the name/mouseover text for the user achievement page can be found here: 

https://trello.com/c/F4MwDe9D/3-joined-a-guild-badge

### PR DESCRIPTION
**Pull request by CooperFLe**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Working on #8614
### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

So far, I have the badge working, but the user must log out and back in the see the badge after earning it. 

It was suggested that I need to sync the client user with the server, but I don't think that is the case because of where I added the code. The party on and party up badges are in the same place and work just fine. 

I'm hoping posting this PR will help someone help me figure out my problem. I'm probably just completely overlooking something obvious. Sorry to ask for help when I'm supposed to be the one helping out here!

[//]: # (Put User ID in here - found in Settings -> API)

----
UUID:

### CODE REVIEW

File: test/api/v3/integration/groups/POST-groups_invite.test.js [hunk 1]
```diff
@@ -4,9 +4,11 @@ import {
   translate as t,
 } from '../../../../helpers/api-v3-integration.helper';
 import { v4 as generateUUID } from 'uuid';
+import nconf from 'nconf';
```
Comment (paglias) [inline test/api/v3/integration/groups/POST-groups_invite.test.js]: I think you pulled some changes from other PRs

Comment (CooperFLe) [reply test/api/v3/integration/groups/POST-groups_invite.test.js]: That does appear to have happened... How do I undo that...

**Comment by paglias** [comment]
Hmm. You may try to switch to another branch, cherry pick (git cherry-pick
commitNumber) the commits you need and then create a new PR from the new
branch

Il 18 Apr 2017 20:31, "Cooper F Le" <notifications@github.com> ha scritto:

**Comment by CooperFLe** [comment]
@paglias looks like only my changes are part of this PR now.

File: website/server/controllers/api-v3/groups.js [hunk 2]
```diff
@@ -561,6 +561,12 @@ api.joinGroup = {
       }
     }
 
+    if (group.type === 'guild') {
```
Comment (TheHollidayInn) [inline website/server/controllers/api-v3/groups.js:564]: Can you add a tests around this new functionality?

Comment (CooperFLe) [reply website/server/controllers/api-v3/groups.js:564]: I don't actually know how to write tests. Or even where to start for that. If you can point me in the right direction, I'd be happy to give it a shot though.

File: website/server/controllers/api-v3/groups.js [hunk 3]
```diff
@@ -561,6 +561,12 @@ api.joinGroup = {
...
+      promises.push(User.update({
+        'achievements.guild': {$ne: true},
```
Comment (TheHollidayInn) [inline website/server/controllers/api-v3/groups.js:566]: This also will update all users. You will want to limit this to users from this guild only. Actually, I think what you want to do here is update the joining user only. 

So, if (!user. achievements.guild) user. achievements.guild = true or something

File: website/server/controllers/api-v3/groups.js (hunk 2 above)
Comment (TheHollidayInn) [reply website/server/controllers/api-v3/groups.js:564]: You will want to read through these examples: https://github.com/HabitRPG/habitica/blob/develop/test/api/v3/integration/groups/POST-groups_groupId_join.test.js

To run test, use `gulp test:api-v3:integration` from the command line.

It is helpful to note that you can run individual tests by adding `it.only`

**Comment by CooperFLe** [comment]
Alright, this has gotten out of hand. I am not sure where all of these addition commits are coming from, and every time I try to get rid of them more appear. I'm going to close this PR, and get a new one going later hopefully containing ONLY my changes.

**Pull request closed without merging**